Gems/ROS2/Code/Source/Gripper/GripperActionServerComponent.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root
+ * of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "GripperActionServer.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <ROS2/Gripper/GripperRequestBus.h>
+#include <control_msgs/action/gripper_command.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+namespace ROS2
+{
+    //! GripperActionServerComponent is a component that encapsulates gripper action server.
+    //! It is responsible for creating and managing the action server, it is responsible for producing feedback and result.
+    class GripperActionServerComponent
+        : public AZ::Component
+        , private AZ::TickBus::Handler
+    {
+    public:
+        using GripperCommand = control_msgs::action::GripperCommand;
+        using GoalHandleGripperCommand = rclcpp_action::ServerGoalHandle<control_msgs::action::GripperCommand>;
+        AZ_COMPONENT(GripperActionServerComponent, "{6A4417AC-1D85-4AB0-A116-1E77D40FC816}", AZ::Component);
+        GripperActionServerComponent() = default;
+        ~GripperActionServerComponent() = default;
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
```

Comment:
Do we support 2 such components in one entity? If not, fill in GetIncompatibleServices.

Reply:
We don't, corrected